linker.go

```diff
@@ -136,5 +136,22 @@ func fixupJumpsAndCalls(insns asm.Instructions) error {
 		}
 	}
 
+	// fixupBPFCalls replaces bpf_probe_read_{kernel,user}[_str] with bpf_probe_read[_str] on older kernels
+	// https://github.com/libbpf/libbpf/blob/master/src/libbpf.c#L6009
+	iter = insns.Iterate()
```

Comment:
Why is this implemented in an additional loop?

Reply:
IMHO, the above two for-loops are for handling symbol call relocation, and both have if-condition that will filter bpf_probe_read instructions out. Thought it would be cleaner for another for-loop.

Or we can implement here https://github.com/cilium/ebpf/blob/76832385809beed631f0a1b8e4f7c51b6d24fd89/linker.go#L93 , what do you think?

Reply:
Your current approach and putting it in the first loop are both OK. I wouldn't put it in the second loop.

Reply:
Okay, let me mark this as resolved.